src/lib/Containers/Inbox.tsx

```diff
@@ -49,9 +46,8 @@ export class Inbox extends React.Component<Props, State> {
 
     if (this.activeBids && this.conversations) {
       // Allow Conversations & Active Bids to properly force-fetch themselves.
-      // The stored refs are the Relay containers; the components themselves are nested under as refs.
-      this.activeBids.refs.component.refreshActiveBids()
-      this.conversations.refs.component.refreshConversations(() => {
+      this.activeBids.refreshActiveBids()
+      this.conversations.refreshConversations(() => {
```

Comment:
nice, simpler

Reply:
And using public API.

Reply:
Just nitpicking for you guys here: Stuff coming from `ref` callbacks in `React` can be set to null (by the ref callback) ie. when the component is unmounting. I'd advice setting the type to `T | null` and then initialize it to `null`. Also newer TS versions have definite assignment checks that will complain about this sort of code.

You can then define a method such as this:

```ts
function assertNonNull<T>(obj: T | null): T {
  if (obj == null) {
    throw new Error('Object is null');
  }
  return obj;
}
```

And use that when you assume the object is non null. You may be aware of this and simply having chosen to save the hassle, but I thought I'd let you know :)

Reply:
I'm not sure you can type a function to never return if some condition is not met, sadly. I believe it is baked into flow that the invariant function behaves like that.

> Regarding the ref callback also calling for unmounting, that’s just a thing that never happens for these two components.

Yeah I know, I'm just personally a bit obsessed with semantics and there's something in me that itches when I see stuff typed in an incorrect way, as I know the next person editing the code is going to rely on the types and make assumptions based on that. I understand how doing the code properly can be a bit of a hinderance and annoyance though.

Reply:
Ah yes, good point. Looks like it is being discussed https://github.com/Microsoft/TypeScript/issues/10421 & https://github.com/Microsoft/TypeScript/issues/8655.

I hear you on worrying about the next person, but I try to strike a balance for my coworkers between untyped and typed code that doesn’t become too tiresome on them :)

Reply:
The beauty of TypeScript 😄